src/test/java/com/jpprade/jcgmtosvg/CGM4SVGTest.java

```
@@ -0,0 +1,124 @@
+package com.jpprade.jcgmtosvg;
+
+import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import com.jpprade.jcgmtosvg.extension.SVGGraphics2DHS;
+
+import net.sf.jcgm.core.CGMDisplay;
+
+import java.awt.geom.AffineTransform;
+import java.awt.geom.Point2D;
+import java.awt.geom.Rectangle2D;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+class CGM4SVGTest {
+    private CGM4SVG cgm4svg;
+    private SVGPainter mockPainter;
+    private Map<String, Object> options;
+
+    @BeforeEach
+    void setUp() throws IOException {
+        mockPainter = Mockito.mock(SVGPainter.class);
+        options = new HashMap<>();
+        byte[] emptyCgmData = new byte[0];
+        ByteArrayInputStream inputStream = new ByteArrayInputStream(emptyCgmData);
+        cgm4svg = new CGM4SVG(inputStream, mockPainter, options);
+    }
+
+    @Test
+    void testConstructorWithFile() throws IOException {
+        File tempFile = File.createTempFile("test", ".cgm");
+        CGM4SVG cgmFromFile = new CGM4SVG(tempFile, mockPainter);
+        assertNotNull(cgmFromFile);
+    }
+
+    @Test
+    void testConstructorWithInputStream() {
+        assertNotNull(cgm4svg);
+    }
+
+    @Test
+    void testOptionsAreSetCorrectly() {
+        options.put("hotSpotEnabled", true);
+        CGM4SVG newCgm4svg;
+        try {
+            newCgm4svg = new CGM4SVG(new ByteArrayInputStream(new byte[0]), mockPainter, options);
+            assertTrue((boolean) newCgm4svg.options.get("hotSpotEnabled"));
+        } catch (IOException e) {
+            e.printStackTrace();
```

let the test throw IOException instead